core/src/components/button/button.ionic.scss

```diff
@@ -203,283 +201,62 @@
-:host(.button-outline) {
-  --border-color: #{globals.ion-color(primary, base)};
-  --background: #{globals.$ionic-color-base-white};
-  --color: #{globals.ion-color(primary, base)};
-}
-
-// Clear Button
-// --------------------------------------------------
-
-// Default Clear Color
-:host(.button-clear) {
-  --border-width: #{globals.$ionic-border-size-0};
-  --background: transparent;
-  --color: #{globals.$ionic-color-primary-base};
-}
-
-// Block Button
-// --------------------------------------------------
-
-:host(.button-block) {
-  display: block;
-}
-
-:host(.button-block) .button-native {
-  @include globals.margin-horizontal(0);
-
-  width: 100%;
-
-  clear: both;
-
-  contain: content;
-}
-
-:host(.button-block) .button-native::after {
-  clear: both;
-}
-
-// Full Button
-// --------------------------------------------------
-
-:host(.button-full) {
-  display: block;
-}
-
-:host(.button-full) .button-native {
-  @include globals.margin-horizontal(0);
-
-  width: 100%;
-
-  contain: content;
-}
-
-:host(.button-full:not(.button-round)) .button-native {
-  @include globals.border-radius(0);
-
-  border-right-width: 0;
-  border-left-width: 0;
-}
-
-.button-native {
-  @include globals.border-radius(var(--border-radius));
-  @include globals.font-smoothing();
-  @include globals.margin(0);
-  @include globals.padding(var(--padding-top), var(--padding-end), var(--padding-bottom), var(--padding-start));
-  @include globals.text-inherit();
-
-  display: flex;
-
-  position: relative;
-
-  align-items: center;
-
-  width: 100%;
-  height: 100%;
-
-  min-height: inherit;
-
-  transition: var(--transition);
-
-  border-width: var(--border-width);
-  border-style: var(--border-style);
-  border-color: var(--border-color);
-
-  outline: none;
-
-  background: var(--background);
-
-  line-height: 1;
-
-  box-shadow: var(--box-shadow);
-
-  contain: layout style;
-  cursor: pointer;
-
-  opacity: var(--opacity);
-  overflow: var(--overflow);
-
-  z-index: 0;
-  box-sizing: border-box;
-  appearance: none;
-}
-
-.button-native::-moz-focus-inner {
-  border: 0;
-}
-
-.button-inner {
-  display: flex;
-  position: relative;
-
-  flex-flow: row nowrap;
-  flex-shrink: 0;
-  align-items: center;
-  justify-content: center;
-
-  width: 100%;
-  height: 100%;
-
-  z-index: 1;
-}
-
-// Button Slots
-// --------------------------------------------------
-
-::slotted([slot="start"]),
-::slotted([slot="end"]) {
-  flex-shrink: 0;
+  opacity: 0.6;
 }
 
 // Button Icons
 // --------------------------------------------------
 
 ::slotted(ion-icon) {
   font-size: globals.$ionic-font-size-500;
-  pointer-events: none;
-}
-
-// Button: States
-// --------------------------------------------------
-
-.button-native::after {
-  @include globals.button-state();
 }
 
-// Button Focused
-:host(.ion-focused) {
-  color: var(--color-focused);
+::slotted(ion-icon[slot="start"]) {
+  @include globals.margin-horizontal(null, globals.$ionic-space-200);
```

Comment:
It looks like this only applies to the `sm` button, here are the Figma values:

| size | space |
| ---| ---|
| `sm` | `space/200` |
| `md` | `space/250` |
| `lg` | `space/300` |

Reply:
https://github.com/ionic-team/ionic-framework/pull/29754/commits/a638797e22b469a5e55be06443ae22c3506c7058